### PR TITLE
Battle struct macros

### PR DESCRIPTION
**Pull request by PikalaxALT**
I am a masochist who cannot be deterred. This pull request attempts to modernize pokeruby to meet or approach the standard set by [pokeemerald](/pret/pokeemerald). The main focus will be cleaning up ewram.h.

### CODE REVIEW

File: include/ewram.h [hunk 1]
```diff
@@ -2,278 +2,171 @@
...
+#define eContestPaintingBuffers         ((*(struct ContestPaintingBuffers*)(gSharedMem + 0x15E00)))
+#define eEnableContestDebugging         (gSharedMem[0x18000])
+#define eContestTempSave                (*(struct ContestTempSave *)(gSharedMem + 0x18004))
+#define sContest                        (*(struct Contest *)(gSharedMem + 0x19204))
+#define sContestantStatus               ((struct ContestantStatus *)(gSharedMem + 0x19260))
+#define eContestAppealResults           (*(struct ContestAppealMoveResults *)(gSharedMem + 0x192D0))
+#define eContestAI                      ((struct ContestAIInfo *)(gSharedMem + 0x192E4))
+#define eContestExcitement              (*(struct ContestExcitement *)(gSharedMem + 0x19328))
+#define eContestGfxState                ((struct ContestGfxState *)(gSharedMem + 0x19338))
+#define gContestResources__moveAnim     (*(struct ContestMoveAnim *)(gSharedMem + 0x19348))
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/ewram.h:68]: > Double underscore

Intentional

File: src/battle_main.c [hunk 2]
```diff
@@ -722,6 +723,7 @@ void sub_800F298(void)
             u8 taskId;
 
             ResetBlockReceivedFlags();
+            // LinkBattleComputeBattleTypeFlags
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/battle_main.c:726]: The lines that follow were de-inlined in subsequent releases

File: src/contest_util.c [hunk 3]
```diff
@@ -44,89 +44,87 @@ void SetContestTrainerGfxIds(void)
...
 
-void sub_80C4CF8(void)
+void ShouldReadyContestArtist(void)
 {
     if(!gContestFinalStandings[gContestPlayerMonIndex]
-    && gSpecialVar_ContestRank == 3
-    && gUnknown_02038678[gContestPlayerMonIndex] >= 800)
+    && gSpecialVar_ContestRank == CONTEST_RANK_MASTER
+    &&
+        gContestMonTotalPoints[gContestPlayerMonIndex] >= 800)
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/contest_util.c:117]: CLion thinks it's sOoOoOo smart

File: src/credits.c [hunk 4]
```diff
@@ -611,16 +615,16 @@ void task_a_8143D04(u8 taskIdA)
...
+            (gDecompressionBuffer + 0x800)[i] = 0x22;
         for (i = 0; i < 0x800; i++)
-            HALL_OF_FAME_SHEET_2[i] = 0x33;
+            (gDecompressionBuffer + 0x1000)[i] = 0x33;
 
-        EWRAM_1F800[0] = 0;
-        EWRAM_1F800[1] = 0x53FF; // light yellow
-        EWRAM_1F800[2] = 0x529F; // light red
-        EWRAM_1F800[3] = 0x7E94; // light blue
+        ((u16 *)(gDecompressionBuffer + 0x1800))[0] = RGB( 0,  0,  0); // black
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/credits.c:624]: Leaving as-is intentionally for visual alignment

File: src/battle_script_commands.c [hunk 5]
```diff
@@ -2296,5793 +2301,2786 @@ u8 GetBattlerTurnOrderNum(u8 battlerId)
...
+    gBattlescriptCurrInstr++;                   \
+    /*gBattleCommunication[MOVE_EFFECT_BYTE] = 0;*/ \
+    return;                                     \
+}
+
+#define RESET_RETURN                            \
+{                                               \
+    /*gBattleCommunication[MOVE_EFFECT_BYTE] = 0;*/ \
+    return;                                     \
+}
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/battle_script_commands.c:2315]: > Why have this if the reset isn't being done? If it's just for consistency, I'd argue this is worse.

Result of copy/paste directly from Emerald, then revert to match in Ruby

File: src/battle_script_commands.c [hunk 6]
```diff
@@ -2296,5793 +2301,2786 @@ u8 GetBattlerTurnOrderNum(u8 battlerId)
...
             return;
         }
-        else if (StatusChanged == 0)
-            {gBattlescriptCurrInstr++; return;}
+        else if (statusChanged == FALSE)
+        {
+//            gBattleCommunication[MOVE_EFFECT_BYTE] = 0;
+            gBattlescriptCurrInstr++;
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply src/battle_script_commands.c:2597]: Removed the macro instead

File: include/ewram.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (GriffinRichards) [reply include/ewram.h:68]: Why? We don't have this convention in any labels in the other repos.

[the remark this answers is not in the document]
Comment (PikalaxALT) [reply include/ewram.h:68]: Replacement for Emerald's gContestResources->moveAnim

**Pull request merged**